runtime/pkg/logutil/handler.go

```diff
@@ -21,13 +21,15 @@ func (d *BufferedZapCore) Enabled(level zapcore.Level) bool {
 }
 
 func (d *BufferedZapCore) Write(entry zapcore.Entry, fields []zapcore.Field) error {
+	fields = append(d.fields, fields...)
```

Comment:
Won't this suffer from a similar potential issue of overwriting the underlying array of `d.fields` (which is shared)?

Reply:
We can save a redundant map allocation by passing `d.fields` and `fields` separately to `AddZapEntry`, and having `AddZapEntry` combine the two in the map it builds. I.e. refactor to something like `AddZapEntry(entry, d.fields, fields)`.

Reply:
or I can flip the variables to append call like `fields = append(fields, d.fields...)` ?

Reply:
By passing directly to `AddZapEntry` (which allocates a map for it), we would save an allocation. And since this happens for every log, that’s a lot of allocations.

`append(fields, d.fields...)` would be safe, but still do a new underlying allocation in most cases.